src/OpenTelemetry.Instrumentation.StackExchangeRedis/TracerProviderBuilderExtensions.cs

```diff
@@ -68,22 +75,78 @@ public static class TracerProviderBuilderExtensions
 
             AddRedisInstrumentation(
                 builder,
-                connection,
                 sp.GetOptions<StackExchangeRedisCallsInstrumentationOptions>(),
-                configure);
+                configure,
+                out _,
+                connection);
+        });
+    }
+
+    /// <summary>
+    /// Enables automatic data collection of outgoing requests to Redis.
+    /// </summary>
+    /// <param name="builder"><see cref="TracerProviderBuilder"/> being configured.</param>
+    /// <param name="connectionRegistry">Registry to register connections with.</param>
+    /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+    public static TracerProviderBuilder AddRedisInstrumentation(
+        this TracerProviderBuilder builder,
+        out IConnectionRegistry connectionRegistry) => AddRedisInstrumentation(builder, out connectionRegistry, null);
+
+    /// <summary>
+    /// Enables automatic data collection of outgoing requests to Redis.
+    /// </summary>
+    /// <param name="builder"><see cref="TracerProviderBuilder"/> being configured.</param>
+    /// <param name="connectionRegistry">Registry to register connections with.</param>
+    /// <param name="configure">Optional callback to configure options.</param>
+    /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+    public static TracerProviderBuilder AddRedisInstrumentation(
+        this TracerProviderBuilder builder,
+        out IConnectionRegistry connectionRegistry,
+        Action<StackExchangeRedisCallsInstrumentationOptions> configure)
+    {
+        Guard.ThrowIfNull(builder);
+
+        var defaultRegistry = new ConnectionRegistry();
+        connectionRegistry = defaultRegistry;
+
+        if (builder is not IDeferredTracerProviderBuilder deferredTracerProviderBuilder)
+        {
+            return AddRedisInstrumentation(builder, new StackExchangeRedisCallsInstrumentationOptions(), configure, out defaultRegistry.RegistrationCallback);
+        }
+
+        return deferredTracerProviderBuilder.Configure((sp, builder) =>
+        {
+            AddRedisInstrumentation(
+                builder,
+                sp.GetOptions<StackExchangeRedisCallsInstrumentationOptions>(),
+                configure,
+                out defaultRegistry.RegistrationCallback);
         });
     }
 
     private static TracerProviderBuilder AddRedisInstrumentation(
         TracerProviderBuilder builder,
-        IConnectionMultiplexer connection,
         StackExchangeRedisCallsInstrumentationOptions options,
-        Action<StackExchangeRedisCallsInstrumentationOptions> configure)
+        Action<StackExchangeRedisCallsInstrumentationOptions> configure,
+        out Action<IConnectionMultiplexer> registration,
```

Comment:
How about

```suggestion
        out IConnectionRegistry registry,
```

? 

Personally I find the initialization of the registry using `out defaultRegistry.RegistrationCallback` too smart 😉

Reply:
changed in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1139/commits/d688db626c37454828d709eff9a2f8a2a93ed542